test/index.js

```diff
@@ -1708,6 +1764,62 @@ describe('list', function() {
 
   });
 
+  describe('dropLast', function() {
+
+    it('is a binary function', function() {
+      eq(typeof S.dropLast, 'function');
+      eq(S.dropLast.length, 2);
+    });
+
+    it('type checks its arguments', function() {
+      assert.throws(function() { S.dropLast([1, 2, 3]); },
+                    errorEq(TypeError,
+                            '‘dropLast’ requires a value of type Number ' +
+                            'as its first argument; received [1, 2, 3]'));
+
+      assert.throws(function() { S.dropLast(0, null); },
+                    errorEq(TypeError,
+                            'The second argument to ‘dropLast’ ' +
+                            'cannot be null or undefined'));
+    });
+
+    it('returns a Nothing if n is greater than collection length', function() {
+      eq(S.dropLast(6, ['a', 'b', 'c', 'd', 'e']), S.Nothing());
+      eq(S.dropLast(6, 'abcde'), S.Nothing());
+    });
+
+    it('returns a Nothing if n is negative', function() {
+      eq(S.dropLast(-3, ['a', 'b', 'c', 'd', 'e']), S.Nothing());
+      eq(S.dropLast(-0, ['a', 'b', 'c', 'd', 'e']), S.Nothing());
+      eq(S.dropLast(-3, 'abcde'), S.Nothing());
+      eq(S.dropLast(-0, 'abcde'), S.Nothing());
+      // jshint -W053
+      eq(S.dropLast(new Number(-0), ['a', 'b', 'c', 'd', 'e']), S.Nothing());
+      // jshint +W053
+    });
+
+    it('returns an empty collection if n is equal to collection length', function() {
+      eq(S.dropLast(5, ['a', 'b', 'c', 'd', 'e']), S.Just([]));
+      eq(S.dropLast(5, 'abcde'), S.Just(''));
+    });
+
+    it('returns Just the first three elements from the collection', function() {
```

Comment:
We could coalesce this section, the preceding section, and the following section into something as these lines:

```javascript
eq(S.dropLast(0, [1, 2, 3]), S.Just([1, 2, 3]));
eq(S.dropLast(1, [1, 2, 3]), S.Just([1, 2]));
eq(S.dropLast(2, [1, 2, 3]), S.Just([1]));
eq(S.dropLast(3, [1, 2, 3]), S.Just([]));
eq(S.dropLast(4, [1, 2, 3]), S.Nothing());
eq(S.dropLast(0, 'abc'), S.Just('abc'));
eq(S.dropLast(1, 'abc'), S.Just('ab'));
eq(S.dropLast(2, 'abc'), S.Just('a'));
eq(S.dropLast(3, 'abc'), S.Just(''));
eq(S.dropLast(4, 'abc'), S.Nothing());
```

The description of this block would then be a description of the function in general terms, rather than for a specific N.